commonmark/src/main/java/org/commonmark/renderer/text/CoreTextContentNodeRenderer.java

```diff
@@ -68,15 +68,22 @@ public void visit(BlockQuote blockQuote) {
         visitChildren(blockQuote);
         textContent.write('»');
 
-        writeEndOfLine(blockQuote, null);
+        writeEndOfLineIfNeeded(blockQuote, null);
     }
 
     @Override
     public void visit(BulletList bulletList) {
-        bulletListMarker = bulletList.getBulletMarker();
+        if (bulletListHolder != null) {
```

Comment:
What about a bullet list that is nested within a ordered list (or the other way around)? Is that handled correctly?

Reply:
@JinneeJ Tested now with the following input:

```
source = "* a\n  1. one\n  2. two\n* b"
```

Renders as this with the default renderer:

```
* a1. one
2. two
* b
```

Can you fix that?

Reply:
I overlooked it somehow, sorry! 😅 
Updated PR with according tests and fixes.